src/task4feedback/simulator/schedulers/parla/architecture.py

```diff
@@ -671,4 +824,9 @@ def complete(self, scheduler_state: SystemState) -> bool:
                     complete_flag and self.launchable_tasks[device][task_type].empty()
                 )
 
+        reward = scheduler_state.target_exec_time / convert_to_float(
+            scheduler_state.time.scale_to("ms"))
+        print("Total execution time:", scheduler_state.time, " reward:", reward)
+        self.rl_mapper.optimize_model(reward)
```

Comment:
Just to check, this is the only time optimize_model is called right? Between episodes?

Reply:
Right!